src/planet-a/SubdivTriangleMesh.h

```diff
@@ -0,0 +1,487 @@
+
+    return {ChunkLocalSharedId(offset), rte || btm || lft};
+}
+
+//-----------------------------------------------------------------------------
+
+
+struct Chunk
+{
+    SkTriStorage_t m_skeletonTri;
+};
+
+class ChunkedTriangleMeshInfo
+{
+
+public:
+
+    // from wolfram alpha: "sequence 1 3 9 21 45 93"
+    // non-trivial to calculate. ~10 is over the practical limit for
+    // subdivision levels.
+    static inline constexpr std::array<uint16_t, 10> smc_minFansVsLevel =
+    {
+        1, 3, 9, 21, 45, 93, 189, 381, 765, 1533
+    };
+
+    /**
+     * @brief ChunkedTriangleMesh
+     *
+     * @param chunkMax     [in] Max number of chunks
+     * @param subdivLevels [in] Number of times a chunk triangle is subdivided.
+     *                          Practical limit: ~8
+     * @param sharedMax    [in] Max number of shared vertices
+     * @param fanMax       [in] Max number of Fan triangles reserved for a chunk
+     */
+    ChunkedTriangleMeshInfo(uint16_t chunkMax, uint8_t subdivLevels,
+                            uint32_t sharedMax, uint16_t fanMax)
+     : m_chunkMax{ chunkMax }
+     , m_chunkSubdivLevel{ subdivLevels }
+     , m_chunkWidth{ uint16_t(1u << subdivLevels) }
+     , m_chunkIds{ chunkMax }
+     , m_chunkData{ std::make_unique<Chunk[]>(chunkMax) }
+
+     , m_chunkVrtxFillCount{ uint16_t((m_chunkWidth-2) * (m_chunkWidth-1) / 2)}
+
+     , m_chunkVrtxSharedCount{ uint16_t(m_chunkWidth * 3) }
+     , m_chunkSharedUsed{
+           std::make_unique<SharedVrtxStorage_t[]>(chunkMax * m_chunkVrtxSharedCount) }
+
+     , m_chunkIndxFanOffset{ uint32_t(m_chunkWidth)*m_chunkWidth
+                             - smc_minFansVsLevel[subdivLevels] }
+     , m_chunkIndxFillCount{ m_chunkIndxFanOffset + fanMax }
+
+     , m_sharedMax{ sharedMax }
+     , m_sharedIds{ sharedMax }
+     , m_sharedRefCount{ sharedMax }
+     , m_sharedSkVrtx{ std::make_unique<SkVrtxStorage_t[]>(m_sharedMax) }
+     , m_sharedFaceCount{ std::make_unique<uint8_t[]>(m_sharedMax) }
+
+     , m_vrtxSharedOffset{ uint32_t(chunkMax)*m_chunkVrtxFillCount }
+
+     , m_vrtxCountMax{ uint32_t(chunkMax)*m_chunkVrtxFillCount + sharedMax }
+     , m_indxCountMax{ uint32_t(chunkMax) * uint32_t(m_chunkIndxFillCount) }
+    { };
+
+    ChunkId chunk_create(
+            SubdivTriangleSkeleton& rSkel,
+            SkTriId skTri,
+            ArrayView_t<SkVrtxId> const edgeRte,
+            ArrayView_t<SkVrtxId> const edgeBtm,
+            ArrayView_t<SkVrtxId> const edgeLft);
+
+    constexpr VertexId chunk_coord_to_vrtx(
+            ChunkId chunkId, uint16_t x, uint16_t y) const noexcept
+    {
+        if (auto const [localId, shared] = coord_to_shared(x, y, m_chunkWidth);
+            shared)
+        {
+            return shared_get_vrtx(chunk_shared(chunkId)[size_t(localId)]);
+        }
+
+        // Center, non-shared chunk vertex
+        return VertexId(m_chunkVrtxFillCount * size_t(chunkId) + xy_to_triangular(x - 1, y - 2));
```

Comment:
Why bother promoting the uint16_t to a size_t before multiplication?

Reply:
Likely it's something leftover. uint32_t would work instead of size_t. m_chunkVrtxFillCount is also a uint16_t, and I'm kind of paranoid that the multiplication result would be uint16_t as well, and will overflow.